include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1164,12 +1164,12 @@ struct OrtApi {
    * Create a tensor with user's buffer. You can fill the buffer either before calling this function or after.
    * p_data is owned by caller. ReleaseValue won't release p_data.
    *
-   * \param[in] info
-   * \param[in] p_data
-   * \param[in] p_data_len
-   * \param[in] shape
-   * \param[in] shape_len
-   * \param[in] type
+   * \param[in] info Memory description of where the p_data buffer resides (CPU vs GPU etc).
+   * \param[in] p_data Pointer to the data buffer.
+   * \param[in] p_data_len The number of bytes in the data buffer.
+   * \param[in] shape Pointer to the tensor shape values.
+   * \param[in] shape_len The tensor shape length.
```

Comment:
Should we say rank of the tensor? Or number of dimensions in the shape? Length seems confusing.

Reply:
ok. what about the name "shape_len"?

Reply:
yeah, that's confusing too. 'shape' can be renamed to 'dims' and 'shape_len' to 'num_dims'. One example is this https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/c_api.h#L230-L231